app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
+  def check
+    unless logged_in_as_admin?
```

Comment:
Sorry, to clarify, what I mean is we should have `def clean_up_subscriptions(user)` and `def set_works_revised_at_date(user)` etc as private methods in the controller that are not exposed via routes, and then the method `troubleshoot_common` would call all of them. 

Although as I type this, I am thinking we might want to use these methods in more places after all, in which case they probably should go in the user model, and then troubleshoot_common would just call them from admin_users. Anyway, I think that is a debatable question, but I do feel strongly they should be in admin_users and not users.